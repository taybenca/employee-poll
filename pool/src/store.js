import { createStore } from "redux";
import reducer from "./reducers";
import middleware from "./middleware";

export const store = createStore(reducer, middleware);