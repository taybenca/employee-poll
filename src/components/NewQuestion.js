import { useState } from "react";
import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";
import { handleAddQuestion } from "../actions/questions.js";

const NewQuestion = ({ dispatch }) => {
    const navigate = useNavigate();
    const [optionOneText, setOptionOneText] = useState("");
    const [optionTwoText, setOptionTwoText] = useState("");

    const onChangeOptionOne = (e) => {
        setOptionOneText(e.target.value);
    };

    const onChangeOptionTwo = (e) => {
        setOptionTwoText(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        dispatch(handleAddQuestion(optionOneText, optionTwoText));
        setOptionOneText("");
        setOptionTwoText("");
        navigate("/");
    };

    return (
        <div>
            <h2>Would You Rather?</h2>
            <h4>Create Your Own Poll</h4>
    
            <form onSubmit={handleSubmit}>
                <div>
                    <div>
                        First Option:
                        <input
                            type="text"
                            value={optionOneText}
                            placeholder="Option One..."
                            onChange={onChangeOptionOne}
                            className="poll-select"
                        />
                    </div>
                    <div>
                        Second Option:
                        <input
                            type="text"
                            value={optionTwoText}
                            placeholder="Option Two..."
                            onChange={onChangeOptionTwo}
                            className="poll-select"
                        />
                    </div>
                </div>
                <div>
                    <button
                        className="button"
                        disabled={optionOneText === "" || optionTwoText === ""}
                    >
                    Add question
                    </button>
                </div>
            </form>
        </div>
    );
};
    
export default connect()(NewQuestion);
