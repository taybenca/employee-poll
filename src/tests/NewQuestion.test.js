import { fireEvent, render } from '@testing-library/react'
import * as React from 'react'
import NewQuestion from '../components/NewQuestion'
import { Provider } from "react-redux";
import { BrowserRouter, Form } from "react-router-dom";
import { store } from "../store.js";

describe('NewQuestion', () => {
    it ('expected match snapshot', () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <NewQuestion />
                </BrowserRouter>
            </Provider>
        )
        expect(component).toMatchSnapshot()
    })

    it ('expected change first option when user type', () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <NewQuestion />
                </BrowserRouter>
            </Provider>
        )

        const optionOne = component.getByTestId("option-one")
        fireEvent.change(optionOne, {target: {value: 'Red'}})
        
        expect(optionOne.value).toEqual('Red')
    })

    it ('expected change second option when user type', () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <NewQuestion />
                </BrowserRouter>
            </Provider>
        )

        const optionTwo = component.getByTestId("option-two")
        fireEvent.change(optionTwo, {target: {value: 'Blue'}})
        
        expect(optionTwo.value).toEqual('Blue')
    })

    it ('find the button in the document', () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <NewQuestion />
                </BrowserRouter>
            </Provider>
        )
        
        const button = component.getByTestId('submit-question')
        expect(button).toBeInTheDocument()
    })
})