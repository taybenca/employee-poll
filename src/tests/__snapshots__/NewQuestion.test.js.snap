// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQuestion expected match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2>
          Would You Rather?
        </h2>
        <h4>
          Create Your Own Poll
        </h4>
        <form>
          <div>
            <div>
              First Option:
              <input
                class="poll-select"
                data-testid="option-one"
                placeholder="Option One..."
                type="text"
                value=""
              />
            </div>
            <div>
              Second Option:
              <input
                class="poll-select"
                data-testid="option-two"
                placeholder="Option Two..."
                type="text"
                value=""
              />
            </div>
          </div>
          <div>
            <button
              class="button"
              data-testid="submit-question"
              disabled=""
            >
              Add question
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2>
        Would You Rather?
      </h2>
      <h4>
        Create Your Own Poll
      </h4>
      <form>
        <div>
          <div>
            First Option:
            <input
              class="poll-select"
              data-testid="option-one"
              placeholder="Option One..."
              type="text"
              value=""
            />
          </div>
          <div>
            Second Option:
            <input
              class="poll-select"
              data-testid="option-two"
              placeholder="Option Two..."
              type="text"
              value=""
            />
          </div>
        </div>
        <div>
          <button
            class="button"
            data-testid="submit-question"
            disabled=""
          >
            Add question
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
